src/mlpack/methods/reinforcement_learning/worker/one_step_q_learning_worker.hpp

```diff
@@ -0,0 +1,172 @@
+        learningNetwork.Parameters().n_cols);
+
+    // Construct the local network from global network.
+    NetworkType network = learningNetwork;
+
+    // Get initial state.
+    StateType state = environment.InitialSample();
+
+    size_t steps = 0;
+    double totalReturn = 0.0;
+
+    // Store pending transitions for update later on.
+    std::vector<TransitionType> pending;
+
+    while (!stop && (!config.StepLimit() || steps < config.StepLimit())) {
+      // Interact with the environment.
+      arma::colvec actionValue;
+      network.Predict(state.Encode(), actionValue);
+      ActionType action = policy.Sample(actionValue, deterministic);
+      StateType nextState;
+      double reward = environment.Sample(state, action, nextState);
+      bool terminal = environment.IsTerminal(nextState);
+
+      totalReturn += reward;
+      steps++;
+
+      if (deterministic)
+      {
+        if (terminal)
+          break;
+        else
+        {
+          state = nextState;
+          continue;
+        }
+      }
+
+      #pragma omp atomic
+      totalSteps++;
+
+      pending.push_back(std::make_tuple(state, action, reward, nextState));
+
+      // Do update.
+      if (terminal || pending.size() >= config.UpdateInterval()) {
+        // Initialize the gradient storage.
+        arma::mat totalGradients(learningNetwork.Parameters().n_rows,
+            learningNetwork.Parameters().n_cols);
+        for (size_t i = 0; i < pending.size(); ++i) {
+          TransitionType &transition = pending[i];
+
+          // Compute the target state-action value.
+          arma::colvec actionValue;
+          #pragma omp critical
+          {
+            targetNetwork.Predict(
+                std::get<3>(transition).Encode(), actionValue);
+          };
+          double targetActionValue = actionValue.max();
+          if (terminal && i == pending.size() - 1)
+            targetActionValue = 0;
+          targetActionValue = std::get<2>(transition) +
+              config.Discount() * targetActionValue;
+
+          // Compute the training target for current state.
+          network.Forward(std::get<0>(transition).Encode(), actionValue);
+          actionValue[std::get<1>(transition)] = targetActionValue;
+
+          // Compute gradient.
+          arma::mat gradients;
+          network.Backward(actionValue, gradients);
+
+          // Accumulate gradients.
+          totalGradients += gradients;
+        }
+
+        // Clamp the accumulated gradients.
+        totalGradients = arma::clamp(totalGradients,
+            -config.GradientLimit(), config.GradientLimit());
+
+        // Perform async update of the global network.
+        updater.Update(learningNetwork.Parameters(),
+            config.StepSize(), totalGradients);
+
+        // Sync the local network with the global network.
+        network = learningNetwork;
+
+        pending.clear();
+      }
+
+      if (terminal)
+        break;
+      else
```

Comment:
You mean we can't write:

```
if (terminal)
  break;

state = nextState;
```

instead of:

```
if (terminal)
  break;
else
  state = nextState;
```